app/models/memberships/switch_stammsektion.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+#  Copyright (c) 2024, Schweizer Alpen-Club. This file is part of
+#  hitobito_sac_cas and licensed under the Affero General Public License version 3
+#  or later. See the COPYING file at the top-level directory or at
+#  https://github.com/hitobito/hitobito_sac_cas
+
+module Memberships
+  class SwitchStammsektion < MemberJoinSectionBase
+
+    validate :assert_join_date
+
+    private
+
+    def save_roles
+      Role.transaction do
+        terminate_then_create_roles_for(affected_people)
```

Comment:
Das weicht eigentlich von der Spezifikation ab. In der Spez wird eigentlich schon in `#prepare_roles` die bestehende Rolle terminiert.
Habe das mit @daniel-illi besprochen und laut ihm ist das trotzdem ok so, wollte es aber als fyi hier noch hinterlegen